app/src/main/java/com/nextlevelstudy/services/base/AuthInterceptor.java

```diff
@@ -0,0 +1,62 @@
+package com.nextlevelstudy.services.base;
+
+import android.content.SharedPreferences;
+
+import androidx.annotation.NonNull;
+
+import com.nextlevelstudy.R;
+
+import java.io.IOException;
+
+import javax.inject.Inject;
+import javax.inject.Singleton;
+
+import okhttp3.HttpUrl;
+import okhttp3.Interceptor;
+import okhttp3.Request;
+import okhttp3.Response;
+
+/**
+ * This class add information (API Key) to {@link okhttp3.OkHttpClient} which is passed in
+ * {@link com.nextlevelstudy.di.module.NetworkModule#okHttpClient(HttpLoggingInterceptor)}
+ * which is required when making a request. This will ensure that all requests are made with the API key
+ *
+ */
+
+@Singleton
+public class AuthInterceptor implements Interceptor {
+
+  private String jwtToken;
+
+  /**
+   * Default constructor.
+   */
+  @Inject
+  public AuthInterceptor() {}
+
+  public void setJwtToken(String jwtToken) {
+    this.jwtToken = jwtToken;
+  }
+
+  @Override
+  public Response intercept(@NonNull Chain chain) throws IOException {
+    Request original = chain.request();
+
+    if (original.url().encodedPath().toLowerCase().contains("/users/login") &&
```

Comment:
Instead of hardcoding these rules in the interceptor, you can just split up network calls into different web services and apply the interceptor only on those that do need it.